### DISCUSSION TITLE
brainstorming a 1.0 vision

### DISCUSSION THREAD

**Discussion by progrium** [progrium/darwinkit#10] [Ideas]
At time of writing, macdriver is v0.0.1 with no releases. I use 0.0.x for early private development and usually release 0.1.0 when made public. So I'll be doing that soon. However, I'd like to also start defining a clear path to a 1.0 release. Please share any thoughts or ideas here.

**Comment by TotallyGamerJet** [comment]
I think the best thing to do would be to create a standard for writing bindings. What types are acceptable as input and output and how are they translated from the original to Go syntax. For example, anywhere an NSString is used functions can only take and return go strings? This document can also explain how you plan on handling methods that take a protocol. Should all protocols be represented by interfaces that are then converted to OBJC classes before invocation? I think the best thing is to really on Go's static typing as much as possible because currently, it's too easy to accidentally call the wrong method or send the wrong object to functions.

**Reply by progrium** [reply]
yes, this is would be great. a guide for adding new bindings. if anybody wants to get it started or stub it out in the wiki that would be helpful.

**Comment by progrium** [comment]
Going to table this since there's a lot of things to figure out with minor releases before we'd be close to a 1.0
